spec/unit/classes/server/config_spec.rb

```diff
@@ -34,8 +34,9 @@
           class { 'postgresql::globals':
             manage_package_repo => true,
             version => '9.4',
-          }->
-          class { 'postgresql::server': }
+            before => Class['Postgresql::Server'],
+          }
+          include postgresql::server
```

Comment:
@DavidS - OK after 4.6.1 is released I will re-submit the pull request with a single commit and make the changes you suggest. Let me know if there is a way to inline the expression in the selector that will work in Ruby 1.9 & Puppet 3.